include/seqan3/core/algorithm/bound.hpp

```diff
@@ -6,15 +6,18 @@
 // -----------------------------------------------------------------------------------------------------
 
 /*!\file
- * \brief Provides seqan3::lower_bound and seqan3::upper_bound.
+ * \brief [DEPRECATED] Provides seqan3::lower_bound and seqan3::upper_bound.
  * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ * \deprecated This header will be removed in 3.1.
  */
 
 #pragma once
 
 #include <seqan3/core/concept/core_language.hpp>
 #include <seqan3/core/detail/strong_type.hpp>
 
+SEQAN3_DEPRECATED_HEADER("This header is deprecated and will be removed in SeqAn-3.1.0.")
```

Comment:
Here as well. We need the SEQAN3_DEPRECATED_310 macro before the strong type declarations and update the documentation of the classes with a deprecation notice.